java/src/apps/configurexml/PerformActionModelXml.java

```diff
@@ -57,12 +63,36 @@ public boolean load(Element shared, Element perNode) {
         String className = shared.getAttribute("name").getValue();
         PerformActionModel model = new PerformActionModel();
         model.setClassName(className);
-        for (Element child : shared.getChildren("property")) { // NOI18N
+        shared.getChildren("property").forEach(child -> { // NOI18N
+            String value = child.getAttributeValue("value"); // NOI18N
             if (child.getAttributeValue("name").equals("systemPrefix") // NOI18N
-                    && child.getAttributeValue("value") != null) { // NOI18N
-                model.setSystemPrefix(child.getAttributeValue("value")); // NOI18N
+                    && value != null) {
+                // handle the situation where the model expects a system prefix
+                // but was not saved with one in a pre-4.19.7 JMRI instance
+                // TODO: at some point (June 2022 release?) change entire
+                // try/catch block to just "model.setSystemPrefix(value);"
+                try {
+                    Class<?> ac = Class.forName(className);
+                    if (value.isEmpty() && SystemConnectionAction.class.isAssignableFrom(ac)) {
+                        SystemConnectionAction<?> a = (SystemConnectionAction<?>) ac.getConstructor().newInstance();
+                        InstanceManager.getList(SystemConnectionMemo.class)
+                                .forEach(memo -> a.getSystemConnectionMemoClasses().stream()
+                                .filter(mc -> memo.getClass().isAssignableFrom(mc))
+                                .forEach(mc -> model.setSystemPrefix(memo.getSystemPrefix())));
+                    } else {
+                        model.setSystemPrefix(value);
+                    }
+                } catch (ClassNotFoundException
+                        | InstantiationException
+                        | IllegalAccessException
+                        | IllegalArgumentException
+                        | InvocationTargetException
+                        | NoSuchMethodException
+                        | SecurityException ex) {
+                    // ignore to allow manager to handle later
```

Comment:
Would it make sense to log (at WARN? INFO?) this occurrence so that people trying to solve a downstream problem (i.e. should the Manager _not_ handle it; I'm not sure they will absent tests) get a hit to look back to hear?  (2 places)